error[E0432]: unresolved import `std::backtrace` when building with rustc 1.38.0-dev

Hi,
I get following errors if I try to compile anyhow 1.0.12 with rustc 1.38.0 itself compiled in `dev` channel:
```bash
error[E0432]: unresolved import `std::backtrace`
 --> src/backtrace.rs:2:21
  |
2 | pub(crate) use std::backtrace::Backtrace;
  |                     ^^^^^^^^^ could not find `backtrace` in `std`

error[E0432]: unresolved import `std::backtrace`
 --> src/context.rs:7:10
  |
7 | use std::backtrace::Backtrace;
  |          ^^^^^^^^^ could not find `backtrace` in `std`
```
```bash
strohel@edgy ~/projekty/anyhow $ rustc --version
rustc 1.38.0-dev
```

It seems to be caused by the fact that `cargo:rustc-cfg=backtrace` is errorneously enabled in `build.rs`. If I do a simple change
```diff
diff --git a/build.rs b/build.rs
index dcc19c6..65541b4 100644
--- a/build.rs
+++ b/build.rs
@@ -34,6 +34,6 @@ fn rustc_version() -> Option<Compiler> {
     };
 
     Some(Compiler {
-        nightly: version.contains("nightly") || version.contains("dev"),
+        nightly: version.contains("nightly"),
     })
 }
```
then the crate compiles and tests pass (but that probably isn't the right thing to do).

Perhaps presence of `std::backtrace` can be tested more directly (maybe even without `build.rs`?).

--- a/build.rs
+++ b/build.rs
@@ -1,6 +1,6 @@
 use std::env;
 use std::process::Command;
-use std::str;
+use std::str::{self, FromStr};
 
 fn main() {
     let compiler = match rustc_version() {
@@ -8,32 +8,28 @@ fn main() {
         None => return,
     };
 
-    if compiler.nightly {
+    if compiler.minor >= 40 && compiler.nightly {
         println!("cargo:rustc-cfg=backtrace");
     }
 }
 
 struct Compiler {
+    minor: u32,
     nightly: bool,
 }
 
 fn rustc_version() -> Option<Compiler> {
-    let rustc = match env::var_os("RUSTC") {
-        Some(rustc) => rustc,
-        None => return None,
-    };
+    let rustc = env::var_os("RUSTC")?;
+    let output = Command::new(rustc).arg("--version").output().ok()?;
+    let version = str::from_utf8(&output.stdout).ok()?;
 
-    let output = match Command::new(rustc).arg("--version").output() {
-        Ok(output) => output,
-        Err(_) => return None,
-    };
+    let mut pieces = version.split('.');
+    if pieces.next() != Some("rustc 1") {
+        return None;
+    }
 
-    let version = match str::from_utf8(&output.stdout) {
-        Ok(version) => version,
-        Err(_) => return None,
-    };
-
-    Some(Compiler {
-        nightly: version.contains("nightly") || version.contains("dev"),
-    })
+    let next = pieces.next()?;
+    let minor = u32::from_str(next).ok()?;
+    let nightly = version.contains("nightly") || version.contains("dev");
+    Some(Compiler { minor, nightly })
 }

